tests/ValueObjectTest.php

```diff
@@ -1,184 +1,157 @@
+        self::assertSame($amount, $a->getValue());
+        self::assertSame((string) $amount, (string) $a);
     }
 
     /**
      * @dataProvider invalidAmountProvider
      *
      * @param mixed $amount
-     * @return void
      */
-    public function testAmountInvalidValues($amount)
+    public function testAmountInvalidValues($amount): void
     {
-        $this->setExpectedException('InvalidArgumentException', 'Value has to be an integer.');
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('Value has to be an integer.');
+
         Amount::create($amount);
     }
 
     /**
      * @dataProvider currencyCodeProvider
-     *
-     * @param string $currency
-     * @return void
      */
-    public function testCurrencyCode($currency)
+    public function testCurrencyCode(string $currency): void
     {
         $a = CurrencyCode::create($currency);
         $b = new CurrencyCode($currency);
 
-        static::assertTrue($a->equals($b));
-        static::assertTrue($b->equals($a));
-        static::assertSame($currency, $a->getValue());
-        static::assertSame((string) $currency, (string) $a);
+        self::assertTrue($a->equals($b));
+        self::assertTrue($b->equals($a));
+        self::assertSame($currency, $a->getValue());
+        self::assertSame($currency, (string) $a);
     }
 
     /**
      * @dataProvider invalidCurrencyCodeProvider
      *
      * @param mixed $currency
-     * @return void
      */
-    public function testCurrencyCodeInvalidValues($currency)
+    public function testCurrencyCodeInvalidValues($currency): void
     {
-        $this->setExpectedException('InvalidArgumentException', 'Value `' . $currency . '` is not valid ISO 4217 currency code');
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('Value `' . $currency . '` is not valid ISO 4217 currency code');
+
         CurrencyCode::create($currency);
     }
 
     /**
      * @dataProvider identifierProvider
      *
-     * @param string $id
-     * @return void
+     * @param mixed $id
      */
-    public function testIdentifier($id)
+    public function testIdentifier($id): void
     {
         $a = Identifier::create($id);
         $b = new Identifier($id);
 
-        static::assertTrue($a->equals($b));
-        static::assertTrue($b->equals($a));
-        static::assertSame((string) $id, $a->getValue());
-        static::assertSame((string) $id, (string) $a);
+        self::assertTrue($a->equals($b));
+        self::assertTrue($b->equals($a));
+        self::assertSame((string) $id, $a->getValue());
+        self::assertSame((string) $id, (string) $a);
     }
 
     /**
      * @dataProvider invalidIdentifierProvider
-     *
-     * @param string $id
-     * @return void
      */
-    public function testIdentifierInvalidValues($id)
+    public function testIdentifierInvalidValues(string $id): void
```

Comment:
Tento test má podle mě zbytečný dataprovider, dělá jednoduchou věc, jen zkontroluje, že při vstupu velkého řetězce vrátí konkrétní hlášku, to nemá smysl komplikovat rovnou bych to upravil na:
```php
    public function testIdentifierInvalidValues(): void
    {
        $this->expectException(\InvalidArgumentException::class);
        $this->expectExceptionMessage('Value\'s length has to be up to 100 characters');
        Identifier::create(self::randomChars(101));
    }
```
A pak ten dataprovider zruš.